File: schemas/helpdesk-customer-support/convex/sla_policy_targets.ts
// sla_policy_targets: response and resolution time targets per priority within an SLA policy.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const sla_policy_targets = defineTable({
  slaPolicyId: v.id("sla_policies"),
  priorityId: v.id("ticket_priorities"),
  firstResponseMinutes: v.optional(v.number()),
  nextResponseMinutes: v.optional(v.number()),
  resolutionMinutes: v.optional(v.number()),
})
  .index("by_sla_policy_id_and_priority_id", ["slaPolicyId", "priorityId"]);

File: schemas/helpdesk-customer-support/spacetimedb/sla_policy_targets.rs
// sla_policy_targets: Response and resolution time targets per priority within an SLA policy.
// See README.md for full design rationale.

#[spacetimedb::table(name = sla_policy_targets, public)]
pub struct SlaPolicyTarget {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub sla_policy_id: String, // UUID — FK → sla_policies.id (cascade delete)
    pub priority_id: String, // UUID — FK → ticket_priorities.id (cascade delete)
    // Composite unique: (sla_policy_id, priority_id) — not supported inline, enforce in application
    pub first_response_minutes: Option<i32>,
    pub next_response_minutes: Option<i32>,
    pub resolution_minutes: Option<i32>,
}
